src/flag_gems/ops/eq.py

```diff
@@ -13,7 +13,14 @@ def eq_func(x, y):
 
 def eq(A, B):
     logging.debug("GEMS EQ")
-    O = eq_func(A, B)
+    if A.dim() == 0 and B.dim()==0:
+        O = eq_func_scalar(A.item(), B.item())
+    elif A.dim() == 0:
```

Comment:
can't 0-dim tensor broadcast to high-dimension?

Reply:
pointwise_dynamic can deal with 0-dim tensor. while the problem here is that equal may receive a cuda tensor and a cpu tensor. the solution is cast cpu tensor to the same device as cuda tensor.